src/darwin/Framework/CHIP/MTRDeviceController_Concrete.h

```diff
@@ -36,9 +38,8 @@ typedef void (^MTRDeviceConnectionCallback)(MTRBaseDevice * _Nullable device, NS
 @class MTRCommissionableBrowserResult;
 @class MTRSetupPayload;
 @protocol MTRDevicePairingDelegate;
-@protocol MTRDeviceControllerDelegate;
 
-@interface MTRDeviceController_Concrete : MTRDeviceController
+@interface MTRDeviceController_Concrete : MTRDeviceController <MTRDeviceControllerDelegate>
```

Comment:
I'm probably missing something, but why is the controller a delegate of itself? (This seems odd... maybe it's late)

Reply:
Because we pass it as the MTRDeviceControllerDelegate to the SDK bridge.  The controller then just fans out the notifications to the actual delegates.